.github/actions/make/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Fetch or make an artifact
+description: Fetch an artifact with a given key. If it doesn't exist, run the given command and upload it.
+inputs:
+  key:
+    description: The artifact key
+    required: true
+  make-rule:
+    description: The GNU make rule which produces the artifact
+    required: true
+  target-path:
+    description: The relative path of the final artifact(s)
+    required: false
```

Comment:
I think `target-path` should be required to make it explicit for every job that uses this action what it actually stores.